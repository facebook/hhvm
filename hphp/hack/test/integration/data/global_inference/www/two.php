<?hh //partial

class Two extends One {
  public function foo() {
    return 2;
  }

  public function bar() {
    return 1;
  }

  public function usenum() {
    $x = new One();

    $x->noparameter(1);
  }
}

function throwException() {
  throw new \Exception("");
}

function throwExceptionIndirect() {
  throwException();
}
