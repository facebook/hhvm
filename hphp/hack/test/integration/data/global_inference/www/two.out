<?hh //partial

class Two extends One {
  public function foo() : <<__Soft>> int {
    return 2;
  }

  public function bar() : <<__Soft>> int {
    return 1;
  }

  public function usenum() : <<__Soft>> void {
    $x = new One();

    $x->noparameter(1);
  }
}

function throwException() {
  throw new \Exception("");
}

function throwExceptionIndirect() : <<__Soft>> void {
  throwException();
}
